src/crimson/net/SocketMessenger.cc

```diff
@@ -129,7 +128,7 @@ seastar::future<> SocketMessenger::start(Dispatcher *disp) {
   return seastar::now();
 }
 
-seastar::future<crimson::net::ConnectionXRef>
+seastar::future<crimson::net::ConnectionRef>
```

Comment:
so there is no need to return a future anymore, as we can (or have to) return the `ConnectionRef` right away on current shard, right?

Reply:
Agree, will change in a separate commit.

Reply:
done.